src/java/fr/paris/lutece/plugins/extend/web/component/ResourceExtenderComponentManager.java

```diff
@@ -1,395 +1,405 @@
+    /** The Constant BEAN_MANAGER. */
+    public static final String BEAN_MANAGER = "extend.resourceExtenderComponentManager";
+
+    // CONSTANTS
+    private static final String ORDER_BY_DATE_CREATION = " date_creation ";
+
+    // PROPERTIES
+    private static final String PROPERTY_DEFAULT_LIST_HISTORIES_PER_PAGE = "extend.listHistories.itemsPerPage";
+
+    // PARAMETERS
+    private static final String PARAMETER_ID_EXTENDER = "idExtender";
+    private static final String PARAMETER_FROM_URL = "from_url";
+    private static final String PARAMETER_MANAGE_BY_RESOURCE = "manageByResource";
+
+    // MARKS
+    private static final String MARK_RESOURCE_EXTENDER = "resourceExtender";
+    private static final String MARK_LOCALE = "locale";
+    private static final String MARK_RESOURCE_EXTENDER_CONFIG = "resourceExtenderConfig";
+    private static final String MARK_RESOURCE_EXTENDER_INFO = "resourceExtenderInfo";
+    private static final String MARK_LIST_HISTORIES = "listHistories";
+    private static final String MARK_PAGINATOR = "paginator";
+    private static final String MARK_NB_ITEMS_PER_PAGE = "nb_items_per_page";
+    private static final String MARK_FROM_URL = "from_url";
+    private static final String MARK_EXTENDER_TYPE_MODIFY_CONFIG = "extenderTypeModifyConfig";
+    private static final String MARK_RESOURCE_EXTENDER_CONFIGURABLE = "configurableExtenderTypes";
+
+    // TEMPLATES
+    private static final String TEMPLATE_RESOURCE_EXTENDER_CONFIG = "admin/plugins/extend/resource_extender_config.html";
+    private static final String TEMPLATE_RESOURCE_EXTENDER_DEFAULT_CONFIG = "admin/plugins/extend/resource_extender_default_config.html";
+    private static final String TEMPLATE_RESOURCE_EXTENDER_INFO = "admin/plugins/extend/resource_extender_info.html";
+    private static final String TEMPLATE_RESOURCE_EXTENDER_HISTORY = "admin/plugins/extend/resource_extender_history.html";
+
+    // JSP
+    private static final String JSP_URL_VIEW_EXTENDER_HISTORY = "jsp/admin/plugins/extend/ViewExtenderHistory.jsp";
+
+    // MESSAGES
+    private static final String MESSAGE_STOP_GENERIC_MESSAGE = "extend.message.stop.genericMessage";
+
+    // CONSTANT
+    private static final String CONSTANT_AND = "&";
+    private static final String CONSTANT_AND_HTML = "%26";
+
+    // SERVICES
+    @Inject
+    private IResourceExtenderService _resourceExtenderService;
+    @Inject
+    private IResourceExtenderHistoryService _resourceHistoryService;
+
+    // VARIABLES
+    private int _nItemsPerPage;
+    private int _nDefaultItemsPerPage;
+    private String _strCurrentPageIndex;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IResourceExtenderComponent getResourceExtenderComponent( String strExtenderType )
+    {
+        if ( StringUtils.isNotBlank( strExtenderType ) )
+        {
+            List<IResourceExtenderComponent> resourceExtenderComponentList = CDI.current( ).select( IResourceExtenderComponent.class ).stream( )
```

Comment:
It is always better to use injection in a CDI bean and avoid using CDI.current().